### PR TITLE
Add AutoLayoutSizeSource

### PR DESCRIPTION
**Pull request by Tylerian**
A SizeSource that calculates a view size based on its auto-layout constraints.

### CODE REVIEW

File: Sources/Other/SizeSource.swift [hunk 1]
```diff
@@ -47,3 +48,22 @@ open class ClosureSizeSource<Data>: SizeSource<Data> {
...
   }
 }
+
+open class AutoLayoutSizeSource<Data, View: UIView>: SizeSource<Data> {
+  private let dummyView: View
+  private let dummyViewUpdater: ClosureViewUpdateFn<Data, View>
+  private let horizontalFittingPriority: UILayoutPriority
+  private let verticalFittingPriority: UILayoutPriority
+  
+  public init(dummyView: View.Type, horizontalFittingPriority: UILayoutPriority = .defaultHigh, verticalFittingPriority: UILayoutPriority = .defaultLow, viewUpdater: @escaping ClosureViewUpdateFn<Data, View>) {
```
Comment (lkzhao) [inline Sources/Other/SizeSource.swift]: separate this line.

File: Sources/Other/SizeSource.swift [hunk 2]
```diff
@@ -47,3 +48,22 @@ open class ClosureSizeSource<Data>: SizeSource<Data> {
...
+    self.dummyView = View()
+    self.dummyViewUpdater = viewUpdater
+    self.horizontalFittingPriority = horizontalFittingPriority
+    self.verticalFittingPriority = verticalFittingPriority
+  }
+  
+  open override func size(at index: Int, data: Data, collectionSize: CGSize) -> CGSize {
+    self.dummyViewUpdater(self.dummyView, data, index)
+    return self.dummyView.systemLayoutSizeFitting(collectionSize, withHorizontalFittingPriority: self.horizontalFittingPriority, verticalFittingPriority: self.verticalFittingPriority)
```
Comment (lkzhao) [inline Sources/Other/SizeSource.swift]: ditto

**Comment by Tylerian** [comment]
Okay, I've rebased and separated the class on its own file. 
Now I think it's good to go! 
Wdyt @lkzhao?

**Comment by lkzhao** [comment]
This is great, thanks a lot.

**Pull request merged**